board: npcx: add npcx9m6f_evb board.

MulinChao opened the pull request:
This PR is the last one for porting npcx9 series driver to Zephyr. It still has the dependency with [PR34795](https://github.com/zephyrproject-rtos/zephyr/pull/34795) and [PR35287](https://github.com/zephyrproject-rtos/zephyr/pull/35287). Once they were landed, I will do rebasing to pass the sanity checks. 

So far, the driver implementation has passed the following driver/kernel test suites on npcx9m6f_evb board.
pwm
gpio
adc 
watchdog
i2c
uart
kernel/timer

Signed-off-by: Mulin Chao <mlchao@nuvoton.com>

MulinChao committed:
soc: npcx: add support for npcx9 series in ecst.py script.

This CL adds the support for npcx9m3f and npcx9m6f chips in ecst.py
script which generates the NPCX firmware header.

Signed-off-by: Mulin Chao <mlchao@nuvoton.com>

MulinChao committed:
board: npcx: add npcx9m6f_evb board.

Introduce the npcx9m6f_evb board driver. NPCX9M6F_EVB board is a
development platform to evaluate the Nuvoton NPCX9 series
microcontrollers. This board needs to be mated with part number
NPCX996F.

It also includes:
 1. Add CONFIG_BOARD_NPCX9M6F_EVB definition for adc test suite.
 2. Add npcx7m6f_evb.overlay file for gpio test suite.

Signed-off-by: Jun Lin <CHLin56@nuvoton.com>
Signed-off-by: Mulin Chao <mlchao@nuvoton.com>

keith-zephyr commented on boards/arm/npcx9m6f_evb/doc/index.rst:
```diff
@@ -0,0 +1,98 @@
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
```
The board DTS file also enables the following components:
PSL, PWM, ADC, eSPI, I2C, TACH
Plus power management is also supported.

MulinChao replied:
Done.

keith-zephyr commented on boards/arm/npcx9m6f_evb/npcx9m6f_evb.yaml:
```diff
@@ -0,0 +1,21 @@
+  - gnuarmemb
+ram: 64
+flash: 256
+supported:
```
This list should match the doc file.

MulinChao replied:
Done.

sjg20 commented on boards/arm/npcx9m6f_evb/doc/index.rst:
```diff
@@ -0,0 +1,112 @@
+
+This board comes with a Cortex ETM port which facilitates tracing and debugging
+using a single physical connection. In addition, it comes with sockets for
+JTAG only sessions.
```
JTAG-only

MulinChao replied:
Done.

sjg20 reviewed (changes requested):
in commit message:

"This CL introduces " can be "Introduce the "

MulinChao commented:
The failure of" Compliance" is caused by the dependency of [PR34795](https://github.com/zephyrproject-rtos/zephyr/pull/34795) and [PR35287](https://github.com/zephyrproject-rtos/zephyr/pull/35287). Once they're landed, I will do rebasing to fix it.

MulinChao commented:
The failures of sanity checks are caused by the dependency of #35287. Once it was landed, I will do rebasing to fix it.

MulinChao committed:
tests: kernel: gen_isr_table: Add workaround for npcx9 series.

Both NPCX7/9 uses the IRQs at the end of the vector table, for example,
the IRQ 60 and 61 used for Multi-Input Wake-Up Unit (MIWU) devices by
default, and conflicts with ISR used for testing.

This CL changes TEST_NUM_IRQS (The value is changed from 46 to 44) to
move IRQ used for this test suite from 42 to 40 which is reserved in
both NPCX7 and NPCX9 series to resolve the issue.

Signed-off-by: Mulin Chao <mlchao@nuvoton.com>
